packages/langium-vscode/src/language-server/grammar-workspace-manager.ts

```diff
@@ -0,0 +1,57 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import ignore, { Ignore } from 'ignore';
+import { CancellationToken, WorkspaceFolder } from 'vscode-languageserver-protocol';
+import { URI, Utils } from 'vscode-uri';
+import { LangiumSharedServices } from 'langium/src/services';
+import { ConfigurationProvider } from 'langium/src/workspace/configuration';
+import { FileSystemNode } from 'langium/src/workspace/file-system-provider';
+import { DefaultWorkspaceManager } from 'langium/src/workspace/workspace-manager';
+import { LangiumGrammarLanguageMetaData } from 'langium/src/grammar/generated/module';
+
+const CONFIG_KEY = 'build';
+
+interface WorkspaceManagerConf {
+    /**
+     * gitignore separated exclusion patterns, separated by comma
+     */
+    ignorePatterns: string
+}
+
+export class LangiumGrammarWorkspaceManager extends DefaultWorkspaceManager {
+
+    protected readonly configurationProvider: ConfigurationProvider;
+    protected matcher: Ignore | undefined;
+
+    constructor(services: LangiumSharedServices) {
+        super(services);
+        this.configurationProvider = services.workspace.ConfigurationProvider;
+    }
+
+    async initializeWorkspace(folders: WorkspaceFolder[], cancelToken = CancellationToken.None): Promise<void> {
+        const buildConf: WorkspaceManagerConf = await this.configurationProvider.getConfiguration(LangiumGrammarLanguageMetaData.languageId, CONFIG_KEY);
+        const ignorePatterns = buildConf.ignorePatterns?.split(',')?.map(pattern => pattern.trim())?.filter(pattern => pattern.length > 0);
+        this.matcher = ignorePatterns ? ignore().add(ignorePatterns) : undefined;
+        return super.initializeWorkspace(folders, cancelToken);
+    }
+
+    protected includeEntry(entry: FileSystemNode, fileExtensions: string[]): boolean {
+        if (this.matcher) {
+            const parentWsFolder = this.folders?.map(folder => folder.uri).find(folderUri => entry.uri.toString().startsWith(folderUri));
+            let entryPath = entry.uri.path;
+            if (parentWsFolder) {
+                // create path relative to workspace folder root: /user/foo/workspace/entry.txt -> entry.txt
+                const wsFolderPath = URI.parse(parentWsFolder).path + '/';
+                entryPath = entryPath.replace(wsFolderPath, '');
```

Comment:
See my [previous comment](https://github.com/langium/langium/pull/519#discussion_r944211799). Could you try using the [path module](https://nodejs.org/dist/latest-v16.x/docs/api/path.html) instead of `startsWith` and `replace` string operations?

Reply:
Well, it doesn't matter which API we would use, we can't recover the missing information about the WorkspaceFolder we are operating on. That means that we can't create a WSFolder relative path. 
I think the best is to pass over the WorkspaceFolder, I changed the implementation accordingly